### PR TITLE
Fix warnings 2008, 2000, 1834, 2016

### PR DESCRIPTION
**Pull request by azabbasi**
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2008

### CODE REVIEW

File: iothub/device/src/InternalClient.cs [hunk 1]
```diff
@@ -999,6 +999,9 @@ internal async Task OnMethodCalledAsync(MethodRequestInternal methodRequestInter
                 methodResponseInternal = new MethodResponseInternal(methodRequestInternal.RequestId, (int)MethodResponseStatusCode.BadRequest);
 
                 await SendMethodResponseAsync(methodResponseInternal, methodRequestInternal.CancellationToken).ConfigureAwait(false);
+
+                methodResponseInternal?.Dispose();
```
Comment (drwill-ms) [inline iothub/device/src/InternalClient.cs:1003]: Won't this already happen on 1016 in the finally clause?

File: iothub/device/src/InternalClient.cs [hunk 2]
```diff
@@ -1058,6 +1062,8 @@ internal async Task OnMethodCalledAsync(MethodRequestInternal methodRequestInter
 
             await SendMethodResponseAsync(methodResponseInternal, methodRequestInternal.CancellationToken).ConfigureAwait(false);
 
+            methodResponseInternal?.Dispose();
```
Comment (drwill-ms) [inline iothub/device/src/InternalClient.cs:1065]: If `SendMethodResponseAsync` throws, this won't be disposed.

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/src/InternalClient.cs (hunk 1 above)
Comment (azabbasi) [reply iothub/device/src/InternalClient.cs:1003]: Correct. Fixed.

**Pull request merged**